interactive.c

```diff
@@ -168,28 +226,70 @@ void interactiveShowData(void) {
             }
 
             if (trackDataValid(&a->position_valid)) {
-                snprintf(strLat, 8,"%7.03f", a->lat);
-                snprintf(strLon, 9,"%8.03f", a->lon);
+                snprintf(strLat, sizeof(strLat), "%7.03f", a->lat);
+                snprintf(strLon, sizeof(strLon), "%8.03f", a->lon);
             }
 
             if (trackDataValid(&a->airground_valid) && a->airground == AG_GROUND) {
-                snprintf(strFl, 7," grnd");
+                snprintf(strFl, sizeof(strFl), "grnd ");
             } else if (Modes.use_gnss && trackDataValid(&a->altitude_geom_valid)) {
-                snprintf(strFl, 7, "%5dH", convert_altitude(a->altitude_geom));
+                snprintf(strFl, sizeof(strFl), "%5dH", convert_altitude(a->altitude_geom));
             } else if (trackDataValid(&a->altitude_baro_valid)) {
-                snprintf(strFl, 7, "%5d ", convert_altitude(a->altitude_baro));
+                snprintf(strFl, sizeof(strFl), "%5d ", convert_altitude(a->altitude_baro));
+            }
+            double signalDisplay = 10.0 * log10(signalAverage);
+
+
+            if ((Modes.bUserFlags & MODES_USER_LATLON_VALID) && trackDataValid(&a->position_valid)) {
+                distance = greatcircle(Modes.fUserLat, Modes.fUserLon,
+                    a->lat, a->lon);
+
+                distance /= 1000.0;
+
+                distance *= distance_units_conversion;
+                if (distance > distanceMax) {
+                    distanceMax = distance;
+                }
+                snprintf(strDistance, sizeof(strDistance), "%5.1f ", distance);
+                bearing = get_bearing(Modes.fUserLat, Modes.fUserLon, a->lat, a->lon);
+                snprintf(strBearing, sizeof(strBearing), "%5.0f ", bearing);
+            }
+
+            if (signalDisplay > signalMax) {
+                signalMax = signalDisplay;
             }
+            if (signalDisplay < signalMin) {
+                signalMin = signalDisplay;
+            }
+            signalMean += signalDisplay;
```

Comment:
This is going to be the geometric mean (since you're working in dB); not sure if that's what you intended?

Reply:
Yeah, I wanted it to be the simple mean of the displayed values otherwise it would take a whole bunch of explaining to folks. :)